examples/faq/are_you_there.py

```diff
@@ -0,0 +1,65 @@
+from confluent_kafka import Producer, KafkaError
+
+
+class ConfluentException(Exception):
+    """
+        Converts KafkaError into an exception for handling by an application.
+
+        :param KafkaError kafka_error: The KafkaError to convert.
+        :param str msg: Optional Error message override.
+    """
+    def __init__(self, kafka_error, msg=None):
+        self.code = kafka_error.code()
+
+        if msg is None:
+            msg = kafka_error.str()
+
+        self.message = msg
+
+
+class DeliveryReporter(object):
+    __slots__ = ["success"]
+
+    def __init__(self):
+        self.success = False
+
+    def __call__(self, err, msg):
+        """
+        Handles per message delivery reports indicating success or failed delivery attempts.
+
+        :param err: KafkaError indicating message delivery failed.
+        :param msg: The message delivered to the broker.
+        """
+        if err:
+            self.error(err)
+
+        self.success = True
+        print("Message {} delivered!".format(msg.offset()))
+
+    def error(self, err):
+        """
+        Handler for global/generic and message delivery errors.
```

Comment:
This seems to be mixing per-message status propagation with global error notifications, I'm not sure the latter fits.

Also, since this call is not really used for any delivery reports in this example I'm not sure it is warranted.

Keep things minimal.

Reply:
The delivery report callback is actually used to set the success or failure of a delivered message. That flag is then used to decide if we have ever connected before or not. In an attempt to keep things minimal I did not include the actual producing of a message. I can do so to more fully convey how you would use such a tactic.